useful-scripts/start.ts

```diff
@@ -0,0 +1,471 @@
+      port: number;
+      dockerServices?: ServiceSchemaDockerServices;
+    }
+  | {
+      type: 'docker';
+      service: string;
+    }
+  | {
+      type: 'command';
+      command: string | ((options: ProgramOptions) => string);
+      dockerServices?: ServiceSchemaDockerServices;
+    }
+);
+
+type ServiceSchemaDockerServices =
+  | DockerService[]
+  | ((options: ProgramOptions) => DockerService[]);
+
+// Annoyingly, need to define these separately from schemas,
+// or we run into various circular reference issues in the types.
+const allowedServiceNames = [
+  'admin',
+  'content',
+  'data',
+  'frontend',
+  'frontendProd',
+  'processor',
+  'publisher',
+  'notifier',
+  'idp',
+  'db',
+  'dataStorage',
+] as const;
+
+type ServiceName = (typeof allowedServiceNames)[number];
+
+const serviceSchemas: Record<ServiceName, ServiceSchema> = {
+  admin: {
+    root: 'src/GovUk.Education.ExploreEducationStatistics.Admin',
+    colour: chalk.green,
+    type: 'dotnet',
+    dockerServices() {
+      return fs.existsSync(
+        path.join(projectRoot, this.root, 'appsettings.Idp.json'),
+      )
+        ? ['db', 'data-storage']
+        : ['db', 'data-storage', 'idp'];
+    },
+  },
+  content: {
+    root: 'src/GovUk.Education.ExploreEducationStatistics.Content.Api',
+    colour: chalk.cyan,
+    type: 'dotnet',
+    dockerServices: ['db', 'data-storage'],
+  },
+  data: {
+    root: 'src/GovUk.Education.ExploreEducationStatistics.Data.Api',
+    colour: chalk.magenta,
+    type: 'dotnet',
+    dockerServices: ['db', 'data-storage'],
+  },
+  frontend: {
+    root: 'src/explore-education-statistics-frontend',
+    command: 'pnpm dev',
+    colour: chalk.greenBright,
+    type: 'command',
+  },
+  frontendProd: {
+    root: 'src/explore-education-statistics-frontend',
+    command(options) {
+      return options.skipBuild ? 'pnpm start' : 'pnpm build && pnpm start';
+    },
+    colour: chalk.greenBright,
+    type: 'command',
+  },
+  processor: {
+    root: 'src/GovUk.Education.ExploreEducationStatistics.Data.Processor',
+    colour: chalk.rgb(255, 158, 165),
+    port: 7071,
+    type: 'func',
+    dockerServices: ['db'],
+  },
+  publisher: {
+    root: 'src/GovUk.Education.ExploreEducationStatistics.Publisher',
+    colour: chalk.yellow,
+    port: 7072,
+    type: 'func',
+    dockerServices: ['db', 'data-storage'],
+  },
+  notifier: {
+    root: 'src/GovUk.Education.ExploreEducationStatistics.Notifier',
+    colour: chalk.blue,
+    port: 7073,
+    type: 'func',
+    dockerServices: ['data-storage'],
+  },
+  idp: {
+    root: 'src',
+    service: 'idp',
+    colour: chalk.gray,
+    type: 'docker',
+  },
+  db: {
+    root: 'src',
+    service: 'db',
+    colour: chalk.blue,
+    type: 'docker',
+  },
+  dataStorage: {
+    root: 'src',
+    service: 'data-storage',
+    colour: chalk.green,
+    type: 'docker',
+  },
+};
+
+type ProgramOptions = ReturnType<(typeof program)['opts']>;
+
+const program = new Command()
```

Comment:
Minor comment: Just wondering if it's worth adding a unit test for the start script in the future (potentially when more test resource is available). Just thinking this could potentially reduce the scariness of making a change to a script that everyone uses on a day to day basis